libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationRoute.java

```diff
@@ -76,8 +86,24 @@ static Builder builder(Context context, LocaleUtils localeUtils) {
    * @param callback a RetroFit callback which contains an onResponse and onFailure
    * @since 0.5.0
    */
-  public void getRoute(Callback<DirectionsResponse> callback) {
-    mapboxDirections.enqueueCall(callback);
+  public void getRoute(final Callback<DirectionsResponse> callback) {
+    elapsedTimeBuilder.start();
+
+    mapboxDirections.enqueueCall(new Callback<DirectionsResponse>() {
+      @Override
+      public void onResponse(Call<DirectionsResponse> call, Response<DirectionsResponse> response) {
+        elapsedTimeBuilder.end();
+        callback.onResponse(call, response);
+        if (response.body().routes() != null && !response.body().routes().isEmpty()) {
```

Comment:
@devotaaabel should we remove this check?  Or do we only want to send the even if we have routes in the response?

Reply:
Per slack convo, we're only going to send this event when there are routes in the request